aicsimageio/writers/writer.py

```diff
@@ -1,49 +1,51 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
 from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Union
+
+import dask.array as da
+import numpy as np
 
-from .. import constants, types
+from ..constants import Dimensions
+
+###############################################################################
 
 
 class Writer(ABC):
-    def __init__(self, file_path: types.PathLike):
+    """
+    A small class to build standardized image writer functions.
+    """
+
+    @staticmethod
+    @abstractmethod
+    def save(
```

Comment:
should it take a metadata etree or string? Or a secondary save function with metadata, save_with_metadata() or ... ?

Reply:
Was discussed previously in the PR. This is left to be defined per writer as part of the `kwargs`. So `OmeTiffWriter` could potentially accept an `ElementTree` or a compiled `OMEXML` object or something else. A `ZarrWriter` could potentially accept a `Dict` or something but it is entirely up to the writer.

Reply:
https://github.com/AllenCellModeling/aicsimageio/pull/98#discussion_r403255527

Reply:
Even if aicsimageio dealt with some "open" metadata interchange format, whether it's one that we invent or not, each writer would have to be able to convert to its own preferred format.  There's also the read-write scenario where user reads file, manipulates contents, and wishes to write back to file while preserving the metadata.